src/rust/catnap/futures/pop.rs

```diff
@@ -31,34 +25,43 @@ use ::std::{
     },
 };
 
-//==============================================================================
+//======================================================================================================================
 // Constants
-//==============================================================================
+//======================================================================================================================
 
 /// Maximum Size for a Pop Operation
 const POP_SIZE: usize = 9216;
 
-//==============================================================================
+//======================================================================================================================
 // Structures
-//==============================================================================
+//======================================================================================================================
 
 /// Pop Operation Descriptor
 pub struct PopFuture {
     /// Associated queue descriptor.
     qd: QDesc,
     /// Underlying file descriptor.
     fd: RawFd,
+    /// Source socket address.
+    sockaddr: libc::sockaddr_in,
+    /// Receiving buffer.
+    buf: DemiBuffer,
 }
 
-//==============================================================================
+//======================================================================================================================
 // Associate Functions
-//==============================================================================
+//======================================================================================================================
 
 /// Associate Functions for Pop Operation Descriptors
 impl PopFuture {
     /// Creates a descriptor for a pop operation.
     pub fn new(qd: QDesc, fd: RawFd) -> Self {
-        Self { qd, fd }
+        Self {
+            qd,
+            fd,
+            sockaddr: unsafe { mem::zeroed() },
```

Comment:
Is this unsafe block necessary?

Reply:
Yes it is: https://doc.rust-lang.org/std/mem/fn.zeroed.html